documentation/USAGE.rst

```diff
@@ -342,6 +350,64 @@ you need to do something with the entire solution, you can use
         Different solvers, different preconditioners, or a less restrictive
         tolerance may help.
 
+.. _THREADS_VS_RANKS:
+
+OpenMP Threads vs. MPI Ranks
+============================
+
+By default, Trilinos spawns as many OpenMP_ threads as there are cores
+available.  This may very well be an intentional optimization, where
+Trilinos is designed to have one MPI_ rank per node of a cluster, so each
+of the child threads would help with computation but would not compete for
+I/O resources during ghost cell exchanges and file I/O. However, Python's
+`Global Interpreter Lock`_ (GIL) binds all of the child threads to the same
+core as their parent!  So instead of improving performance, each core
+suffers a heavy overhead from the managing those idling threads.
+
+The solution to this is to force Trilinos to use only one OpenMP_ thread::
+
+   $ export OMP_NUM_THREADS=1
+
+Because this environment variable affects all processes launched in the
+current session, you may prefer to restrict its use to :term:`FiPy` runs::
+
+   $ OMP_NUM_THREADS=1 mpirun -np {# of processors} python myScript.py --trilinos
+
+The difference can be extreme.  We have observed the :term:`FiPy` test
+suite to run in `just over two minutes`_ when ``OMP_NUM_THREADS=1``,
+compared to `over an hour and 23 minutes`_ when OpenMP_ threads are
+unrestricted. We don't know why, but `other platforms`_ do not suffer the
+same degree of degradation.
+
+Conceivably, allowing Trilinos unfettered access to OpenMP_ threads with no
+MPI_ communication at all could perform as well or better than purely MPI_
+parallelization.  The plot below demonstrates this is not the case.  We
+compare solution time vs number of OpenMP_ threads for fixed number of
+slots for a `Method of Manufactured Solutions Allen-Cahn problem`_.
+OpenMP_ threading always slows down FiPy performance.
+
+.. plot:: documentation/pyplots/threadanalyze.py
```

Comment:
This isn't showing up for me. My current setup is what's outlined [here](https://github.com/usnistgov/fipy/blob/develop/nix/build.nix#L40). Do I need to install anything else?

Reply:
Is the documentation build part of the tests yet?

Reply:
`matplotlib` and `pandas` are both required. I've added that to the docs.

Reply:
No, the docs aren't tested, yet. That's *after* this ridiculously delayed release goes out.

Reply:
This is still not showing up for me after installing Pandas. I do get one error `WARNING: convert command 'convert' cannot be run.check the image_converter setting`. Is that anything? Don't worry about this though. The script to generate the plot works for me outside of sphinx.

Reply:
The 32 slots values don't have error bars. Are they just really small?

Reply:
I only ran those once because I didn't want to wait for the resources when I was trying to do the benchmark